<?xml version='1.0'?>
<launch>
	<arg name="model_name" default="mantis_uav"/>
	<arg name="frame_id" default="map"/>
	<arg name="rviz" default="false"/>

	<node pkg="mantis_kinematics" type="mantis_forward_kinematics_node" name="mantis_fk" clear_params="true" output="screen">
		<param name="frame_id" value="$(arg frame_id)" />
		<param name="model_id" value="$(arg model_name)" />

		<param name="do_viz" value="false" />
		<param name="end_effector_pose" value="true" />

		<remap from="/mantis_fk/state/odom" to="/$(arg model_name)/odom"/>
		<remap from="/mantis_fk/state/joints" to="/$(arg model_name)/joint_states"/>

		<remap from="/mantis_fk/visualization" to="/$(arg model_name)/visualization"/>

		<rosparam command="load" file="$(find mantis_description)/config/mantis_sim_params.yaml"/>
	</node>

	<!-- start gazebo client -->
	<group if="$(arg rviz)">
		<node name="rviz" pkg="rviz" type="rviz" respawn="false" output="screen"/>
	</group>
</launch>
